lib/bot.js

```diff
@@ -98,6 +100,24 @@ class Bot {
       this.sendToDiscord(author, to, `*${text}*`);
     });
 
+    this.ircClient.on('join', (channel, nick) => {
+      if (this.disableJoinPartMsgs) return;
+      if (nick === this.nickname && !this.announceSelfJoin) return;
```

Comment:
Personally, I found this useful to have on. If the bot had to reconnect for some reason (e.g. config change, lost connection to IRC), the self join message acts as a ping to others in the channel, that the IRC bridge is re-established and ready.

That said, it's currently only half-useful as disconnects are not mentioned in-channel, meaning users won't be aware that the bridge is suddenly unavailable for whatever reason. ^1

----

^1 This is not obvious in Discord; users don't think to check that the bot is online in the right sidebar